app/shared/controllers/RequestHandler.scala

```diff
@@ -57,12 +57,16 @@ object RequestHandler {
       service: Input => Future[ServiceOutcome[Output]],
       errorHandling: ErrorHandling = ErrorHandling.Default,
       resultCreator: ResultCreator[Input, Output] = ResultCreator.noContent[Input, Output](),
-      auditHandler: Option[AuditHandler] = None
+      auditHandler: Option[AuditHandler] = None,
+      modelHandler: Option[Output => Output] = None
   ) extends RequestHandler {
 
     def handleRequest()(implicit ctx: RequestContext, request: UserRequest[_], ec: ExecutionContext, appConfig: AppConfig): Future[Result] =
       Delegate.handleRequest()
 
+    def withResponseObjectHandling(modelHandler: Output => Output): RequestHandlerBuilder[Input, Output] =
```

Comment:
This would need a matching test as well, so that the Shared coverage doesn't reduce the API coverage level